diff --git a/node_modules/nodebb-plugin-mentions/static/autofill.js b/node_modules/nodebb-plugin-mentions/static/autofill.js
index 7587087..1248da4 100644
--- a/node_modules/nodebb-plugin-mentions/static/autofill.js
+++ b/node_modules/nodebb-plugin-mentions/static/autofill.js
@@ -14,6 +14,7 @@ $(document).ready(function () {
 
 		let slugify;
 		const strategy = {
+			// match variable regex defines what triggers the autocomplete
 			match: /\B@([^\s\n]*)?$/,
 			search: function (term, callback) {
 				require(['composer', 'helpers', 'slugify'], function (composer, helpers, _slugify) {
